src/sentry/utils/pubsub.py

```diff
@@ -1,14 +1,18 @@
 from typing import Any
 
 from arroyo.backends.kafka import build_kafka_producer_configuration
-from confluent_kafka import Producer
+
+from sentry.utils.confluent_producer import get_confluent_producer
 
 
 class KafkaPublisher:
     # XXX(markus): Deprecated. Please use `sentry.utils.arroyo_producer.get_arroyo_producer`.
     def __init__(self, connection: dict[str, Any], asynchronous: bool = True) -> None:
-        self.producer = Producer(
-            build_kafka_producer_configuration(default_config=connection or {})
+        connection = connection or {}
+        if "client.id" not in connection:
+            connection["client.id"] = "sentry.utils.pubsub"
+        self.producer = get_confluent_producer(
```

Comment:
Please roll this out behind a feature flag. This is a change that hits the whole application altogether. If there is any issue (like a performance issue when recording the metrics (see https://github.com/getsentry/arroyo/pull/494#pullrequestreview-3402150400) we could have a large incident).

Please create an option with the rollout percentage and roll it out slowly.

You can create the option here https://github.com/getsentry/sentry/blob/master/src/sentry/options/defaults.py and use option automator to rollout.

See how markus did it here https://github.com/getsentry/sentry/pull/98243/files. You do not need to turn each individual producer on as you are doing the same change everywhere, but a single rollout flag is enough.